### PR TITLE
[MNT] Modifications to Code to Ensure Frontend Compatibility

### LINKED ISSUE #39: Make modifications in the code to ensure frontend compatibilty
Some changes are required so that the frontend can be integrated seamlessly
- Add CORS middleware
- For age order validation the current implementation uses the Python input function to get user confirmation. However, the input function is designed for command-line interfaces and does not work in web-based environments.
The current solution is to remove it completely and address this in a separate issue.

### PR DESCRIPTION
**Pull request by Raya679**
- Closes #39 

Changes proposed in this pull request:

- Added CORS Middleware
- Removed the Python input function to get user confirmation  for age validation.
-  If nothing is mentioned about healthy control, the parameter should be left out (should not be assumed `False`)

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: app/api/validators.py [hunk 1]
```diff
@@ -24,19 +24,10 @@ def validate_age_order(
...
-                f"Sorry but minimum age ({min_age}) can’t be bigger than maximum age ({max_age}). Would you like to see the result for min age {max_age} and max age {min_age}? (please respond with yes/no): "
+            return (
+                "Sorry but minimum age can’t be bigger than maximum age."
             )
 
-            if confirmation.lower() == "yes":
-                filtered_ordered_response["min_age"] = str(max_age)
-                filtered_ordered_response["max_age"] = str(min_age)
-                return filtered_ordered_response
-            else:
```
Comment (rmanaem) [inline app/api/validators.py:35]: Curious, is there a reason we're removing this?

**Review by rmanaem — COMMENTED** [review]
Thanks for the PR @Raya679!
I left one comment that entailed most of the changes please take a look.
P.S. I think `[MNT]` prefix is more appropriate for this PR.

File: app/api/validators.py (hunk 1 above)
Comment (Raya679) [reply app/api/validators.py:35]: Yes,
For age order validation the current implementation uses the Python input function to get user confirmation. However, the input function is designed for command-line interfaces and does not work in web-based environments (i.e in our case the react frontend).

Comment (rmanaem) [reply app/api/validators.py:35]: Would it be possible to make adjustments on the side of UI to make this work?

Comment (Raya679) [reply app/api/validators.py:35]: Hi @rmanaem,
I have created a new issue for this over [here](https://github.com/neurobagel/query-tool-ai/issues/41).

**Review by rmanaem — APPROVED** [review]
:cook:

**Pull request merged**